spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/TemporalConvolution.scala

```diff
@@ -0,0 +1,443 @@
+/*
+ * Licensed to Intel Corporation under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * Intel Corporation licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.nn
+
+import com.intel.analytics.bigdl.nn.abstractnn.{Initializable, TensorModule}
+import com.intel.analytics.bigdl.optim.Regularizer
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.Engine
+
+import scala.concurrent.Future
+import scala.reflect.ClassTag
+
+/**
+ * Applies a 1D convolution over an input sequence composed of nInputFrame frames..
+ *   The input tensor in `forward(input)` is expected to be a 2D tensor
+ *   (`nInputFrame` x `inputFrameSize`) or a 3D tensor
+ *   (`nBatchFrame` x `nInputFrame` x `inputFrameSize`).
+ *
+ * @param inputFrameSize The input frame size expected in sequences given into `forward()`.
+ * @param outputFrameSize The output frame size the convolution layer will produce.
+ * @param kernelW The kernel width of the convolution
+ * @param strideW The step of the convolution in the width dimension.
+ * @param propagateBack Whether propagate gradient back, default is true.
+ * @param wRegularizer instance of [[Regularizer]]
+ *                    (eg. L1 or L2 regularization), applied to the input weights matrices.
+ * @param bRegularizer instance of [[Regularizer]]
+ *                    applied to the bias.
+ * @param initWeight Initial weight
+ * @param initBias Initial bias
+ * @param initGradWeight Initial gradient weight
+ * @param initGradBias Initial gradient bias
+ * @tparam T The numeric type in the criterion, usually which are [[Float]] or [[Double]]
+ */
+class TemporalConvolution[T: ClassTag](
+  val inputFrameSize: Int,
+  val outputFrameSize: Int,
+  val kernelW: Int,
+  val strideW: Int = 1,
+  val propagateBack: Boolean = true,
+  var wRegularizer: Regularizer[T] = null,
+  var bRegularizer: Regularizer[T] = null,
+  val initWeight: Tensor[T] = null,
+  val initBias: Tensor[T] = null,
+  val initGradWeight: Tensor[T] = null,
+  val initGradBias: Tensor[T] = null
+)(implicit ev: TensorNumeric[T]) extends TensorModule[T]  with Initializable {
+
+  val weight: Tensor[T] = Tensor[T](outputFrameSize, inputFrameSize * kernelW)
```

Comment:
The main reason of add `initWeight`, `initBias`, `initGradWeight`, `initGradBias` is to allow two or more layers to share the same weight tensor in `Graph`. So the `weight` variable should be
```scala
val weight = if (initWeight == null) Tensor[T](xxx) else iniWeight
```
Add the same to the other three.